src/language_server_protocol.rs

```diff
@@ -603,6 +609,37 @@ impl LanguageClient {
         Ok(position)
     }
 
+    // moves the cursor to the next or previous diagnostic, depending on the value of direction.
+    pub fn cycle_diagnostics(&self, params: &Value, direction: Direction) -> Result<()> {
+        let filename = self.vim()?.get_filename(params)?;
+        let line = self.vim()?.get_position(params)?.line;
+        let mut diagnostics = self.get(|state| state.diagnostics.clone())?;
+        if let Some(diagnostics) = diagnostics.get_mut(&filename) {
+            if direction == Direction::Next {
+                diagnostics.sort_by(|a, b| a.range.start.line.cmp(&b.range.start.line));
+            } else {
+                diagnostics.sort_by(|a, b| b.range.start.line.cmp(&a.range.start.line));
+            }
+
+            if let Some((_, diagnostic)) = diagnostics.iter_mut().find_position(|it| {
+                if direction == Direction::Next {
+                    it.range.start.line > line
+                } else {
+                    it.range.start.line < line
+                }
+            }) {
+                let target = diagnostic.range.start.line + 1;
+                let _: String = self.vim()?.rpcclient.call("execute", json!([target]))?;
```

Comment:
Is it convenient to jump to proper column as well? Especially helpful when there are more than one diagnostics reported for the same line.

Reply:
Yeah good call!